.github/workflows/coverage.yaml

```
@@ -0,0 +1,160 @@
+name: Evmone Coverage Report
+
+on:
+  pull_request:
+    paths:
+      - 'converted-ethereum-tests.txt' # This triggers the workflow only for changes in file.txt
+
+jobs:
+  evmone-coverage-diff:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache EVMONE
+        uses: actions/cache@v2
+        with:
+          path: ${{ github.workspace }}/evmoneimage
+          key: ${{ runner.os }}-evmone-${{ hashFiles('evmone-hash.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-evmone-
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }} # Checks out the PR branch
+          fetch-depth: 0 # Necessary to fetch all history for diff
```

```suggestion
      - name: Checkout code
        uses: actions/checkout@v3
```
I think this action by default checks out the current PR's branch, so the ref is unnecessary.
Also, we seem to be doing this twice at the start of the steps? (Line 12).

---

there must be 2 separate steps. one is to fetch branches, and enother one to restore cache. 
never used the git cache in this context, but it supposed to keep the docker file and only update it when the hash file is changed.